frontend/src/main/graph.js

```diff
@@ -219,3 +226,77 @@ export function update_graph_size(el, data, config) {
         .attr("width", config.width)
         .attr("height", config.height);
 }
+
+/**
+ * Returns information of the given id
+ *
+ * @param data: data
+ * @param name: String
+ */
+// TODO: Display information somewhere
+export function get_information(data, name){
+    name = name.toUpperCase();
+    const data_nodes = data["nodes"];
+    let name_info = {};
+
+    for(const indx in data_nodes){
+        const current_name = data_nodes[indx];
+        if (current_name["name"]  == name){
+            name_info = current_name;
+        }
+    }
+
+    return name_info;
+}
+
+/**
+ * Updates visualization according to what the user searches
+ *
+ * @param data: data
+ * @param name: String
+ */
+export function update_focused_node(data, name) {
+        name = name.toUpperCase();
+        console.log("entered update focused node")
+        // then for each node check if node is a neighbor; if yes set opacity to 1, if not set to 0
+        const neigh = []
+        const data_edges = data["links"];
+
+        for (const indx in data_edges){
```

Comment:
This constructs the neighbor graph completely from scratch on each call. This is redundant with the `focus_node` implementation in the create graph function and the adjacency data structure created there for the same purpose. Please factor out that code so we have a general neighbor function defined and usable in these two places.